### ISSUE TITLE
2.8.1 can't serialize and deserialize date null (2.8.0 works fine)

### ISSUE THREAD

**Issue by xanscale** [google/gson#1096]
```
W/System.err: com.google.gson.JsonParseException: The date should be a string value
W/System.err:     at com.google.gson.DefaultDateTypeAdapter.read(DefaultDateTypeAdapter.java:96)
W/System.err:     at com.google.gson.DefaultDateTypeAdapter.read(DefaultDateTypeAdapter.java:40)
W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129)
W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220)
W/System.err:     at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
W/System.err:     at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
W/System.err:     at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
```

https://github.com/google/gson/blob/b8f616c939c652b8540c95fa2b377b8c628ef3ff/gson/src/main/java/com/google/gson/DefaultDateTypeAdapter.java#L95

this will trows JsonParseException
```
    if (in.peek() != JsonToken.STRING) {
      throw new JsonParseException("The date should be a string value");
    }
```

this is my conf
`new GsonBuilder().serializeNulls().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").create()`

**Comment by inder123** [comment]
Can you create a PR with a failing test? Thanks.

**Comment by tylerOrMe** [comment]
Having similar issue in serialization.
```
Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create()
gson.toJson(myObjectWhoContainsNullDates)
```
throws `NullPointerException` from Gson 2.8.1
2.8.0 works fine.

**Comment by SergeyTrukhachev** [comment]
Faced the same issue with 2.8.1.
2.8.0 works well.

**Comment by bmarty** [comment]
Hi,

This is my workaround for this bug, waiting for 2.8.2:

    private Gson getGson() {
        // Trick to get the DefaultDateTypeAdatpter instance
        // Create a first instance a Gson
        Gson gson = new GsonBuilder()
                .setDateFormat(DATE_FORMAT_COMPLETE)
                .create();

        // Get the date adapter
        TypeAdapter<Date> dateTypeAdapter = gson.getAdapter(Date.class);

        // Ensure the DateTypeAdapter is null safe
        TypeAdapter<Date> safeDateTypeAdapter = dateTypeAdapter.nullSafe();

        // Build the definitive safe Gson instance
        return new GsonBuilder()
                .setDateFormat(DATE_FORMAT_COMPLETE)
                .registerTypeAdapter(Date.class, safeDateTypeAdapter)
                .create();
    }

Hope it will help

BR

**Comment by rafa0809** [comment]
Nice @bmarty . 
I get the impression the last `.setDateFormat(DATE_FORMAT_COMPLETE)` is not needed. You actually defined within the safeDateTypeAdapter all the details to parse the date.

**Comment by bmarty** [comment]
@rafa0809 yes, indeed!

**Comment by nachodd** [comment]
Facing the same issue, @bmarty  solution didn't work for me. 
Had to downgrade to 2.8.0

**Comment by sudr** [comment]
Should this issue be closed as a result of the fix in #1157 .

**Comment by mohamedmelsayed** [comment]
>  TypeAdapter<Date> safeDateTypeAdapter = dateTypeAdapter.nullSafe();

it worked for me @bmarty 
thank you
